fe/fe-core/src/test/resources/sql/tpch/q12-2.sql

```diff
@@ -36,5 +36,5 @@ TOP-N (order by [[25: L_SHIPMODE ASC NULLS FIRST]])
                     INNER JOIN (join-predicate [1: O_ORDERKEY = 11: L_ORDERKEY] post-join-predicate [null])
                         SCAN (columns[1: O_ORDERKEY, 6: O_ORDERPRIORITY] predicate[null])
                         EXCHANGE SHUFFLE[11]
-                            SCAN (columns[21: L_SHIPDATE, 22: L_COMMITDATE, 23: L_RECEIPTDATE, 25: L_SHIPMODE, 11: L_ORDERKEY] predicate[25: L_SHIPMODE IN (MAIL, SHIP) AND 22: L_COMMITDATE < 23: L_RECEIPTDATE AND 21: L_SHIPDATE < 22: L_COMMITDATE AND 23: L_RECEIPTDATE >= 1994-01-01 AND 23: L_RECEIPTDATE <= 1994-12-31])
+                            SCAN (columns[21: L_SHIPDATE, 22: L_COMMITDATE, 23: L_RECEIPTDATE, 25: L_SHIPMODE, 11: L_ORDERKEY] predicate[25: L_SHIPMODE IN (MAIL, SHIP) AND 22: L_COMMITDATE < 23: L_RECEIPTDATE AND 21: L_SHIPDATE < 22: L_COMMITDATE AND 23: L_RECEIPTDATE >= 1994-01-01 AND 23: L_RECEIPTDATE < 1995-01-01])
```

Comment:
> why these changed, it should not change


For example, in testcase `q12-2.sql`, there is a predicate: `and l_receiptdate < date '1994-01-01' + interval '1' year`

Before this bugfix:
        right child `date '1994-01-01' + interval '1' year` will be evaluated to: `1995-01-01 00:00:00 (ConstantOperator, type = DATETIME)`
        the whole predicate will be optimized by the rule: com.starrocks.sql.optimizer.rewrite.scalar.ReduceCastRule#reduceDateToDatetimeCast - since column `l_receiptdate` is of type `DATE`, then it will match the case LT (line 188) and then be converted to LE: `resultBinaryPredicateOperator = BinaryPredicateOperator.le(castChild, newDate);`
        So this predicate finally becomes: `23: L_RECEIPTDATE <= 1994-12-31`
After this bugfix:
        right child `date '1994-01-01' + interval '1' year` will be evaluated to: `1995-01-01 (ConstantOperator, type = DATE)`
        Its type is DATE, matching the column `l_receiptdate`'s type (DATE), so there is no need to apply the rule mentioned above
        So this predicate will not be converted from LT to LE.
        So this predicate finally becomes: `23: L_RECEIPTDATE < 1995-01-01`
        It eliminates one unnecessary optimization call.

This is why I have to change the results for those testcases. Thanks, please point out any mistakes or concerns.